### ISSUE TITLE
SC2086 - Suggestion causing break; shown as no errors

### ISSUE THREAD

**Issue by Winterhuman** [koalaman/shellcheck#2515]
#### For bugs
- Rule Id: 2086
- My shellcheck version: online
- [ y ] The rule's wiki page does not already cover this
- [ y ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ y ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ y ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

colour_status() { echo -e "$(tput $2)::$(tput sgr0) $(tput $3)$1$(tput sgr0)"; }
stat_echo() { colour_status "$1" "bold setaf 4" "bold"; }
stat_echo "Test..."
```

#### Here's what shellcheck currently says:

```
Did you mean: (apply this, apply all SC2086)
colour_status() { echo -e "$(tput "$2")::$(tput sgr0) $(tput "$3")$1$(tput sgr0)"; }
```

#### Here's what I wanted or expected to see:

No error or non-breaking suggestion. With the suggested changes applied, the script above returns (`bash 5.1.016-1`):

```
./test.sh
tput: unknown terminfo capability 'bold setaf 4'
:: Test...
```

EDIT: `$(tput "bold setaf 4")` seems to be the problem as `tput` doesn't like this, I'm not sure if including an exception just for `tput` is appropriate, let me know what you think.

**Comment by Winterhuman** [comment]
This seems to just be https://github.com/koalaman/shellcheck/issues/380#issue-78507728, in this case it's the multiple arguments as a single variable example that the issue presents. Closing

**Comment by kurahaupo** [comment]
I note that for some versions of `tput`, writing `tput bold setaf 4` may simply be a long-winded way of writing `tput bold`.

Perhaps this is what's intended:
```
tput bold &&
tput setaf 4
```
or as a single command:
```
tput -S <<E
bold
setaf 4
E
```

Either way, I don't see how either the shell or shellcheck is supposed to intuit that you want to split into 2 arrays with differing numbers of elements.

I was there in the 1980's when tput and its library were absolutely essential for getting stuff working portably on many terminal types. Not hardcoding escape sequences was repeatedly born upon us.

But 40 years later things have changed.

TL;DR: tput is now doing more harm than good.

* it doesn't understand advanced features like lockable & scrolling regions (_even though those have been around for 40+ years!_), window titles, mouse click & drag, paste buffers, or hyperlinks;
* it doesn't understand 24-bit colour (aka _TrueColor_);
* one of its supposed great benefits - mnemonics - is pointless when you have to remember numeric codes for colours, which - surprise surprise - are exactly those defined by ANSI and its extensions;
* many of the mnemonics aren't really memory aids at all (who remembers the difference between `setf` and `setaf`?)  Now they've become an extra abstraction layer to learn by rote for no good purpose.
* it doesn't allow you to toggle attributes independently, but rather insists that you need to reset to the default state (`sgr0`) when you want to turn any of them off, and then turn back on the ones you still want. For example, if you have bold red text on an yellow background, and you want to go non-bold but still red text on a yellow background, there's no way to ask for and get `\e[22m` * moreover what you get is a sequence of separate escape sequences, like `\e[0m\e[43m\e[31m` instead of `\e[0;4;31m`;
* exception: if you happen to know that "colour 9" really means "default colour" then you can use that with `setaf` and `setab`, but of course, that doesn't output even a single byte to non-ANSI terminal, leaving it in a non-default colour, making the entire notion of "abstraction" pointless
* it's a separate program, not built into the shell, so it's 1000× slower.

There are no non-emulator terminals left, let alone any non-ANSI terminals. We don't need to pander to them any more.
